build/parseSpec.c

```diff
@@ -797,11 +798,20 @@ static rpmSpec parseSpec(const char *specFile, rpmSpecFlags flags,
 	case PART_PREP:
 	    parsePart = parsePrep(spec);
 	    break;
+	case PART_BUILDREQUIRES:
+	    parsePart = parseSimpleScript(spec, "%generate_buildrequires", &(spec->buildrequires));
+	    break;
 	case PART_BUILD:
+	    parsePart = parseSimpleScript(spec, "%build", &(spec->build));
+	    break;
 	case PART_INSTALL:
+	    parsePart = parseSimpleScript(spec, "%install", &(spec->check));
+	    break;
 	case PART_CHECK:
+	    parsePart = parseSimpleScript(spec, "%check", &(spec->check));
+	    break;
 	case PART_CLEAN:
-	    parsePart = parseBuildInstallClean(spec, parsePart);
+	    parsePart = parseSimpleScript(spec, "%clean", &(spec->clean));
```

Comment:
The parseBuildInstallClean() -> parseSimpleScript() generalisation makes sense regardless of the rest of this, please split that part into a separate commit and push.

Reply:
Yeah, this is a separate patch already. Pushed to master.